llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -2375,6 +2375,33 @@ Value *InstCombiner::getFreelyInvertedImpl(Value *V, bool WillInvertAllUses,
     }
   }
 
+  if (PHINode *PN = dyn_cast<PHINode>(V)) {
+    SmallVector<std::pair<Value *, BasicBlock *>, 8> IncomingValues;
+    for (Use &U : PN->operands()) {
+      BasicBlock *IncomingBlock = PN->getIncomingBlock(U);
+      if (isa<PHINode>(U.get()))
+        return nullptr;
+      Value *NewIncomingVal =
+          getFreelyInvertedImpl(U.get(), /*WillInvertAllUses=*/false,
+                                /*Builder=*/nullptr, DoesConsume, Depth);
+      if (NewIncomingVal == nullptr)
+        return nullptr;
+      // Make sure that we can safely erase the original PHI node.
+      if (NewIncomingVal == V)
+        return nullptr;
```

Comment:
Which test checks this case? I'm wondering if this is something that can happen indirectly.

Reply:
See the test `@test_inv_free_loop`. TBH I don't know why it happens. If the PHI node is used twice, `WillInvertAllUses` should be false.

Reply:
I think this is related to freelyInvertAllUsersOf() used for de-Morgan patterns. I think your check is fine.